synapse/handlers/initial_sync.py

```diff
@@ -226,6 +232,8 @@ async def handle_room(event: RoomsForUser) -> None:
                 d["state"] = await self._event_serializer.serialize_events(
                     current_state.values(),
                     time_now=time_now,
+                    # No need to bundle aggregations for state events.
+                    bundle_aggregations=False,
```

Comment:
This a bunch of places where we do this -- we do a similar check in `serialize_event` (by checking `event.is_state()`, but I think it is worthwhile to keep it here since we *know* that these are state events it is more efficient to just pass the boolean. This might be over-optimizing though and I can remove these if it is clearer!

Reply:
I'm ambivalent

Reply:
actually I'm a bit confused. You seem to have added a bunch of these, and then removed them again in a48bd2ad. What's that about?

Reply:
Right, sorry should have explained that!

1. I initially added them to the places where we only serialize a state event.
2. I then added the check for `is_state()`.
3. I then removed passing the flag for places we were serializing a single event, but left it in places where we serialize many events.

As I said, maybe this is an over-optimization.

Do you think I should just leave them all in?

Reply:
> Do you think I should just leave them all in?

I kinda do, to be honest. I think that would be clearer, and have little effect on performance.

I don't have super-strong feelings though.

Reply:
> I kinda do, to be honest.

wait no, the opposite. Let me try again.

I think you should let `serialize_event` sort it out, and just set `bundle_aggregations` where you need to override the behaviour. I think that would be clearer, and have little effect on performance.

Reply:
> I think you should let `serialize_event` sort it out, and just set `bundle_aggregations` where you need to override the behaviour. I think that would be clearer, and have little effect on performance.

OK. I think that's reasonable. I'll make sure that it is documented to not include them for state events too. 👍  Thanks!

Reply:
Hopefully 2e76df3b023604f2b4e2d77eea895663d62ea35a makes this clear.